src/egl/main/eglcurrent.c

```diff
@@ -72,9 +72,10 @@ static inline void _eglFiniTSD(void)
    if (_egl_TSDInitialized) {
       _EGLThreadInfo *t = _eglGetTSD();
 
-      _egl_TSDInitialized = EGL_FALSE;
       if (t && _egl_FreeTSD)
```

Comment:
Wondering if we should just change the original patch to match something like this:
https://chromium.googlesource.com/chromiumos/overlays/chromiumos-overlay/+/master/media-libs/arc-mesa-img/files/0028-egl-automatically-call-eglReleaseThread-on-thread-te.patch

Reply:
I can take a look at this patch. But maybe Yi J may have a better idea how we can proceed? Is it possible for you to include Yi J here?

Reply:
We did have an internal discussion and rebased to this patch now @downor